RIPS/rip-9997.md

```diff
@@ -0,0 +1,198 @@
+
+The validation frames of the AA_TX_TYPE transaction are represented as individual virtual transactions by the clients.
+They are assigned their own sequential `transactionIndex`, and their `transactionHash` is defined as
+(`AA_TX_TYPE transaction hash + 1`).
+
+All block-related RPC methods, like `eth_getBlockByHash` and `eth_getBlockByNumber`, must include these virtual
+transactions as part of the `transactions` field and include validation in the block transaction count.
+
+All transaction-related RPC methods, like `eth_getTransactionByHash` and `eth_getTransactionReceipt`, must
+accept the virtual transaction hash as input and return the details calculated as if the validation was a
+separate transaction.
+
+There is a number of behaviours that define transaction-wide effects in Ethereum.
+This list includes, but is not limited to:
+
+* Tracking `accessed_addresses`
+* [EIP-1283](https://eips.ethereum.org/EIPS/eip-1283) Gas metering for SSTORE
+* [EIP-1153](https://eips.ethereum.org/EIPS/eip-1153) Transient storage opcodes
+
+Any such behaviour has separate effects in the "Validation Virtual Transaction" and "Execution Transaction".
+
+Gas refunds are issued at the end of the entire transaction only.
+
+### Unused gas penalty charge
+
+```
+UNUSED_GAS_PENALTY = 10
+```
+
+Transactions of type `BUNDLE_TRANSACTION_SUBTYPE` that reserve a lot of gas for themselves using `validationGasLimit`,
+`paymasterGasLimit` and `callGasLimit` fields but do not use the reserved gas present a challenge for
+block builders. This is especially demanding in case a gas used by a transaction can be significantly different
+based on its position within a block, as such transactions may cause the block builder to iterate its algorithm
+many times until a fully utilized block is discovered.
+
+A penalty of `UNUSED_GAS_PENALTY` percent of the entire unused gas limit is charged from the
+transaction `sender` or `paymaster`.
+
+The total gas limit is calculated as `totalLimit = validationGasLimit + paymasterGasLimit + callGasLimit`.\
+The `totalGasUsed` is calculated as a sum of all gas used during the transaction.\
+The unused gas is calculated as `unusedGas = totalLimit - totalGasUsed`.
+
+The effects of allowing transactions to specify unrestricted gas limits is shown on this diagram:
+![](../assets/rip-7560/unused_gas_attack_overview.png)
+
+## Rationale
+
+### Using no-op transaction instead of adding a field to the block payload
+
+While in this particular case there is no execution frame that needs to happen for
+a `BUNDLE_HEADER_SUBTYPE` transaction, in other similar scenarios it is possible we
+will require some execution for a header transaction.
+
+For example, in (RIP-xxxx: Signature Aggregation)[todo] the header transaction contains
+the execution of the aggregted signature verification code.
+
+For consistency with potential future extensions to the RIP-7560 it seems reasonable to
+introduce a no-op transaction subtype.
+
```

Comment:
In addition, it couldn't be represented as one field because a block may contain any number of such bundles.